### PR TITLE
Google internal rules

### PR DESCRIPTION
**Pull request by pq**
Follow-up from: #1068.

See also: #1074.

/cc @bwilkerson @davidmorgan

### CODE REVIEW

**Comment by bwilkerson** [comment]
I don't think that we should publish this unless there is a commitment to maintain it as the internal rule set changes.

If we have such a commitment, then I think that the header comment should be updated to indicate that the rule set is a living artifact that will evolve over time. We might also want a standard copyright notice.

File: README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ you could lint your package like this:
 
     $ dartanalyzer --options analysis_options.yaml .
     
-and see any violations of the `annotate_overrides`, `hash_and_equals`, and `prefer_is_not_empty` rules in the console.  In practice you would probably configure quite a few more rules (the full list is [here][lints]).
+and see any violations of the `annotate_overrides`, `hash_and_equals`, and `prefer_is_not_empty` rules in the console.  In practice you would probably configure quite a few more rules (the full list is [here][lints]).  The evolving list of lints enforced internally at Google may serve as inspiration [here][google-lints].
```
Comment (zoechi) [inline README.md]: Might be just my personal preference but I dislike links with text `here`.  
Better would be a text that tells what's behind the link.
Why not `... rules ([the full list][lints])`

**Comment by pq** [comment]
> I don't think that we should publish this unless there is a commitment to maintain it as the internal rule set changes.

👍 I don't see why we couldn't commit.

> If we have such a commitment, then I think that the header comment should be updated to indicate that the rule set is a living artifact that will evolve over time. We might also want a standard copyright notice.

Good call.

File: README.md (hunk 1 above)
Comment (pq) [reply README.md]: I totally agree.  Thanks!

Comment (bwilkerson) [reply README.md]: My writing style is somewhat different, so the tone might not match, but I'd write something like this:
```
... more rules.

To help you choose the rules you want to enable for your package, we have provided both a [complete list of rules][lints] and a list of the [lints that are enforced internally at Google][google-lints].
```

Comment (zoechi) [reply README.md]: perfectly fine for me as well

Comment (pq) [reply README.md]: Updated to @bwilkerson's suggestion. 👍

**Comment by pq** [comment]
OK, some updates.  Word-smithing welcome.

(Off to find ☕️.)

File: example/google.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+linter:
+  rules:
+    - avoid_empty_else
+    - avoid_return_types_on_setters
+    - avoid_types_as_parameter_names
+    - control_flow_in_finally
+    - no_duplicate_case_values
+    - prefer_contains
+    - prefer_equal_for_default_values
```
Comment (davidmorgan) [inline example/google.yaml]: prefer_is_not_empty landed today :)

**Comment by davidmorgan** [comment]
Looks good, thanks!

Since I'm responsible for updating the internal list, I'm happy to commit to keeping the two in sync. I can also add a bit of color, e.g. rejected rules.

File: example/google.yaml (hunk 2 above)
Comment (pq) [reply example/google.yaml]: 🎉

**Comment by pq** [comment]
> Since I'm responsible for updating the internal list, I'm happy to commit to keeping the two in sync. I can also add a bit of color, e.g. rejected rules.

Awesome.  Thanks a million!

**Pull request merged**